subsys/bluetooth/controller/ll_sw/lll_scan.h

```diff
@@ -31,6 +31,7 @@ struct lll_scan {
 	uint16_t duration_expire;
 	uint8_t  phy:3;
 	uint8_t  is_adv_ind:1;
+	uint8_t  is_aux_prep:1;
```

Comment:
This is not needed, as I believe isr_rx can decide using `lll_aux` if the reception was LLL or ULL scheduled.

Reply:
The issue is that `lll_aux` was used for detection, but it only detects if last received PDU was scheduled from LLL. We want to know whether complete radio event was scheduled from scan or aux ticker (I added a flag because I could not find any existing clean way to do this) since we then decide at the end of radio event whether to resume scan (in case we received LLL scheduled aux as a part of scan radio event) or send an aux done event (in case we received aux outside of scan radio event).

Reply:
> `is_aux_prep` is only used to decide between `isr_done` or `lll_scan_isr_resume` which was earlier done using `lll_aux` which I think should still work. Why do we want to `know whether complete radio event was scheduled from scan or aux ticker` ?

Aux ticker can fire anywhere outside scan window so resuming scan from isr after it was complete does not make much sense imo.

> Split the HCI commit into a separate PR, so that I can approve it. Remaining changes we can continue in this PR.

Will do.